src/pybind/ceph_argparse.py

```diff
@@ -741,7 +741,7 @@ def matchnum(args, signature, partial=False):
                 else:
                     # it was required, and didn't match, return
                     return matchcnt
-        if desc.req:
+        if desc.numseen > 0:
```

Comment:
So, I don't quite understand why or how this works; it seems to me that at this point in the loop a partial match will also have set desc.numseen == 1.

I think the right answer is to only allow partial matching on the final argument.  Something like this:

             try:
-                validate_one(word, desc, partial)
+                # only allow partial matching if we're on the last supplied
+                # word; avoid matching foo bar and foot bar just because
+                # partial is set
+                validate_one(word, desc, partial and (len(words) == 0))